src/algorithm/distance3d.cpp

```diff
@@ -763,7 +763,7 @@ squared_distance_t squaredDistanceSegmentTriangle3D(
     /*
      * If [sAsB] intersects the triangle (tA,tB,tC), distance is 0.0
      */
-    if ( boost::none != CGAL::intersection( sAB, tABC ) ) {
+    if ( CGAL::intersection( sAB, tABC ) ) {
         return 0.0 ;
     }
```

Comment:
It would be better to call `do_intersect` (the predicate) instead of `intersection` (the construction).

Reply:
shall I just replace CGAL::intersection() by CGAL::do_intersect() ?

Reply:
Yes, that should do the work: check if the two objects (a segment and a triangle) intersect.